frontend/src/plugins/impl/FileBrowserPlugin.tsx

```diff
@@ -0,0 +1,279 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import { z } from "zod";
+import { Table, TableCell, TableRow } from "@/components/ui/table";
+import { createPlugin } from "../core/builder";
+import { useState } from "react";
+import { useAsyncData } from "@/hooks/useAsyncData";
+import { rpc } from "../core/rpc";
+import { Checkbox } from "@/components/ui/checkbox";
+import { Folder } from "lucide-react";
+import { toast } from "@/components/ui/use-toast";
+import { NativeSelect } from "@/components/ui/native-select";
+
+/**
+ * Arguments for a file browser component.
+ *
+ * @param initialPath - the path to display on component render
+ * @param filetypes - filter directory lists by file types
+ * @param multiple - whether to allow the user to select multiple files
+ * @param label - label for the file browser
+ * @param restrictNavigation - whether to prevent the user from accessing
+ * directories outside the set path
```

Comment:
i personally put the comments above each params - i think the autocomplete works better (not sure if you hover over `multiple`, if it will give you the documentation